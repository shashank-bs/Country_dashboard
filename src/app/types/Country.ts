export interface Country {
    name: {
      common: string;
      official:string;
    };
    population: number;
    region: string;
    capital: string[];
    flags: {
      svg: string;
    };
    currencies: Record<string, { name: string }>;
    languages: Record<string, string>;
    timezones: string[];
    cca2:string;
  }
  