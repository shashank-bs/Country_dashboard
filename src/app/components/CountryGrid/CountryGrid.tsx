// src/components/CountryGrid.tsx
import { Country } from "../../types/Country";
import { Grid2 } from "@mui/material";
import CountryCard from "../CountryCard/CountryCard";

interface CountryGridProps {
  countries: Country[];
}

const CountryGrid: React.FC<CountryGridProps> = ({ countries }) => {
  return (
    <Grid2 container spacing={2}>
      {countries.map((country) => (
        <Grid2 size={{ xs: 12, sm: 6, md: 4, lg: 3 }} key={country.cca2}>
          <CountryCard country={country} key={country.name.official} />
        </Grid2>
      ))}
    </Grid2>
  );
};

export default CountryGrid;
