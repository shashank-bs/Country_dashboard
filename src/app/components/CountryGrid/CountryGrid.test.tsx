import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import CountryGrid from "./CountryGrid";
import { Country } from "../../types/Country";

describe("CountryGrid Component", () => {
  const mockCountries: Country[] = [
    {
      name: {
        common: "United States",
        official: "United States of America",
      },
      flags: { svg: "https://example.com/flag-us.svg" },
      capital: ["Washington, D.C."],
      population: 331000000,
      region: "Americas",
      currencies: {},
      languages: {},
      timezones: [],
      cca2: "US",
    },
    {
      name: {
        common: "Canada",
        official: "Canada",
      },
      flags: { svg: "https://example.com/flag-ca.svg" },
      capital: ["Ottawa"],
      population: 37742154,
      region: "Americas",
      currencies: {},
      languages: {},
      timezones: [],
      cca2: "CA",
    },
  ];

  test("renders without crashing", () => {
    render(<CountryGrid countries={mockCountries} />);
  });

  test("renders correct number of CountryCard components", () => {
    const { getAllByTestId } = render(
      <CountryGrid countries={mockCountries} />
    );
    expect(getAllByTestId("country-card").length).toBe(mockCountries.length);
  });

  test("renders country names correctly", () => {
    const { getByText } = render(<CountryGrid countries={mockCountries} />);
    mockCountries.forEach((country) => {
      expect(getByText(country.name.common)).toBeInTheDocument();
    });
  });

  test("renders country flags correctly", () => {
    const { getByAltText } = render(<CountryGrid countries={mockCountries} />);
    mockCountries.forEach((country) => {
      expect(
        getByAltText(`Flag of ${country.name.common}`)
      ).toBeInTheDocument();
    });
  });
  test("matches snapshot", () => {
    const { asFragment } = render(<CountryGrid countries={mockCountries} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
