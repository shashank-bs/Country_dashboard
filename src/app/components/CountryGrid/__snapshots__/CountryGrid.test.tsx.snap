// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountryGrid Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row MuiGrid2-spacing-xs-4 css-ngvvop-MuiGrid2-root"
  >
    <div
      class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 MuiGrid2-grid-sm-6 MuiGrid2-grid-md-4 MuiGrid2-grid-lg-3 css-5zitc8-MuiGrid2-root"
    >
      <a
        href="/united%20states"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1rzkyto-MuiPaper-root-MuiCard-root"
          data-testid="country-card"
          style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
        >
          <img
            alt="Flag of United States"
            data-nimg="1"
            decoding="async"
            height="100"
            loading="lazy"
            src="https://example.com/flag-us.svg"
            style="color: transparent; object-fit: cover; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
            width="150"
          />
          <div
            class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          >
            <div
              class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-13y79c5-MuiTypography-root"
            >
              United States
            </div>
            <p
              class="MuiTypography-root MuiTypography-body2 css-atovpb-MuiTypography-root"
            >
              Capital: Washington, D.C.
            </p>
            <p
              class="MuiTypography-root MuiTypography-body2 css-atovpb-MuiTypography-root"
            >
              Population: 33,10,00,000
            </p>
            <p
              class="MuiTypography-root MuiTypography-body2 css-atovpb-MuiTypography-root"
            >
              Region: Americas
            </p>
          </div>
        </div>
      </a>
    </div>
    <div
      class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 MuiGrid2-grid-sm-6 MuiGrid2-grid-md-4 MuiGrid2-grid-lg-3 css-5zitc8-MuiGrid2-root"
    >
      <a
        href="/canada"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1rzkyto-MuiPaper-root-MuiCard-root"
          data-testid="country-card"
          style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
        >
          <img
            alt="Flag of Canada"
            data-nimg="1"
            decoding="async"
            height="100"
            loading="lazy"
            src="https://example.com/flag-ca.svg"
            style="color: transparent; object-fit: cover; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
            width="150"
          />
          <div
            class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          >
            <div
              class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-13y79c5-MuiTypography-root"
            >
              Canada
            </div>
            <p
              class="MuiTypography-root MuiTypography-body2 css-atovpb-MuiTypography-root"
            >
              Capital: Ottawa
            </p>
            <p
              class="MuiTypography-root MuiTypography-body2 css-atovpb-MuiTypography-root"
            >
              Population: 3,77,42,154
            </p>
            <p
              class="MuiTypography-root MuiTypography-body2 css-atovpb-MuiTypography-root"
            >
              Region: Americas
            </p>
          </div>
        </div>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
